ml-agents/mlagents/trainers/trainer/trainer.py

```diff
@@ -116,13 +118,35 @@ def save_model(self, name_behavior_id: str) -> None:
         """
         self.get_policy(name_behavior_id).save_model(self.get_step)
 
-    def export_model(self, name_behavior_id: str) -> None:
+    def export_model(self, name_behavior_id: str, is_checkpoint: bool = False) -> None:
         """
         Exports the model
         """
         policy = self.get_policy(name_behavior_id)
-        settings = SerializationSettings(policy.model_path, policy.brain.brain_name)
-        export_policy_model(settings, policy.graph, policy.sess)
+        if is_checkpoint:
+            checkpoint_path = f"{name_behavior_id}-{self.get_step}"
+            settings = SerializationSettings(
+                policy.model_path, policy.brain.brain_name, checkpoint_path
+            )
+            new_checkpoint: Dict[str, Union[int, str]] = {}
+            # Store steps and file_path
+            new_checkpoint["steps"] = int(self.get_step)
+            new_checkpoint["file_path"] = os.path.join(
+                settings.model_path, f"{settings.checkpoint_path}.nn"
+            )
+            # Record checkpoint information
+            GlobalTrainingStatus.track_checkpoint_info(
+                name_behavior_id, new_checkpoint, policy.keep_checkpoints
+            )
+        else:
+            # Extracting brain name for consistent name_behavior_id
+            brain_name = policy.model_path.split("/")[-1]
```

Comment:
Can't we just use `name_behavior_id`?  I don't understand the problem this is solving.

Reply:
I wasn't exactly sure why name_behavior_id was different for checkpoints vs. final models. When checkpoints are saved, the name_behavior_id was '3DBall', but when the final model was called for saving, the name_behavior_id was '3DBall?team=0'.

Reply:
I think `name_behavior_id` is the right thing to use in both cases.  Let's figure out in which cases the `?team=N` shows up (probably ghost trainer?) and resolve it outside of this method.

Reply:
Adding @andrewcoh

Reply:
The trainer has the name without team id. The policy has the name with team id. The reason for this is that the ghost trainer can manage multiple policies with the same behavior but on different teams.